### PR TITLE
Remove circular dependencies on web

### PR DESCRIPTION
**Pull request by m-bert**
## Description

While working on problems with scroll we discovered that Gesture Handler has many circular dependencies. This PR removes them from web implementation (the only one that's left is related to `hammer`). It also adds `madge` - tool to find circular dependencies in our project.

## Test plan

Run `yarn madge --extensions js,ts,tsx --circular src | grep web` from root.

### CODE REVIEW

File: src/handlers/gestureHandlerCommon.ts [hunk 1]
```diff
@@ -114,6 +113,15 @@ export type ActiveCursor =
   | 'zoom-in'
   | 'zoom-out';
 
+export enum MouseButton {
+  LEFT = 1,
+  RIGHT = 2,
+  MIDDLE = 4,
+  BUTTON_4 = 8,
+  BUTTON_5 = 16,
+  ALL = 31,
```
Comment (tjzel) [inline src/handlers/gestureHandlerCommon.ts:122]: all? 🤯

File: src/web/handlers/GestureHandlerType.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import type { PointerType } from '../../PointerType';
+import type { MouseButton } from '../../handlers/gestureHandlerCommon';
+import type { State } from '../../State';
+import type { Config } from '../interfaces';
+import type EventManager from '../tools/EventManager';
+import type { GestureHandlerDelegate } from '../tools/GestureHandlerDelegate';
+import type PointerTracker from '../tools/PointerTracker';
+
+export default interface GestureHandlerType {
```
Comment (tjzel) [inline src/web/handlers/GestureHandlerType.ts:9]: Terrible name... How about IGestureHandler to signify the class-like contract?

File: src/web/tools/GestureHandlerDelegate.ts [hunk 3]
```diff
@@ -8,10 +7,10 @@ export interface MeasureResult {
   height: number;
 }
 
-export interface GestureHandlerDelegate<T> {
+export interface GestureHandlerDelegate<T, S> {
```
Comment (tjzel) [inline src/web/tools/GestureHandlerDelegate.ts:10]: I would suggest more descriptive type parameters. Expensify does a nice job with prefixing every generic type with the `T` letter. So in here you'd get `TComponent` and `THandler` or something like that.

**Review by tjzel — APPROVED** [review]
Looks good overall!

File: src/handlers/gestureHandlerCommon.ts (hunk 1 above)
Comment (m-bert) [reply src/handlers/gestureHandlerCommon.ts:122]: All 🙆🏾‍♀️

File: src/handlers/createHandler.tsx [hunk 4]
```diff
@@ -9,7 +9,6 @@ import { customDirectEventTypes } from './customDirectEventTypes';
 // @ts-ignore - it isn't typed by TS & don't have definitelyTyped types
 import deepEqual from 'lodash/isEqual';
 import RNGestureHandlerModule from '../RNGestureHandlerModule';
-import type RNGestureHandlerModuleWeb from '../RNGestureHandlerModule.web';
```
[the remark this answers is not in the document]
Comment (m-bert) [reply src/handlers/createHandler.tsx:12]: It was a very long journey 

```
RNGestureHandlerModule.web.ts > web/handlers/FlingGestureHandler.ts >
web/handlers/GestureHandler.ts > web/interfaces.ts >
handlers/gestureHandlerCommon.ts > handlers/handlersRegistry.ts >
handlers/gestures/gesture.ts > handlers/FlingGestureHandler.ts > handlers/createHandler.tsx
```

File: src/web/tools/GestureHandlerDelegate.ts (hunk 3 above)
Comment (m-bert) [reply src/web/tools/GestureHandlerDelegate.ts]: Done in [dbd44e2](https://github.com/software-mansion/react-native-gesture-handler/pull/2783/commits/dbd44e21e969dc6e93f40ea031282b19175bd844).

File: src/web/handlers/GestureHandlerType.ts (hunk 2 above)
Comment (m-bert) [reply src/web/handlers/GestureHandlerType.ts]: Done in [050189e](https://github.com/software-mansion/react-native-gesture-handler/pull/2783/commits/050189e29a3513052eafffd976d89a7a1daa87b6).

**Comment by m-bert** [comment]
Turns out that there was one more dependency introduced in #2788. Right now we should be good to go.

File: src/web/tools/InteractionManager.ts [hunk 5]
```diff
@@ -103,14 +102,15 @@ export default class InteractionManager {
...
-    otherHandler: GestureHandler
+    _handler: IGestureHandler,
+    otherHandler: IGestureHandler
   ): boolean {
-    return (
-      otherHandler instanceof NativeViewGestureHandler &&
-      otherHandler.getState() === State.ACTIVE &&
-      !otherHandler.isButton()
-    );
+    const isNativeHandler =
```
Comment (j-piasecki) [inline src/web/tools/InteractionManager.ts:108]: Please add a comment why it's done this way

File: src/web/tools/InteractionManager.ts [hunk 6]
```diff
@@ -103,14 +102,15 @@ export default class InteractionManager {
...
+      otherHandler.constructor.name === 'NativeViewGestureHandler';
+    const isActive = otherHandler.getState() === State.ACTIVE;
+    const isButton = !!otherHandler.isButton?.();
```
Comment (j-piasecki) [inline src/web/tools/InteractionManager.ts:111]:
```suggestion
    const isButton = otherHandler.isButton?.() === true;
```

**Review by j-piasecki — APPROVED** [review]

File: src/web/tools/InteractionManager.ts (hunk 5 above)
Comment (m-bert) [reply src/web/tools/InteractionManager.ts:109]: Added in [89337e2](https://github.com/software-mansion/react-native-gesture-handler/pull/2783/commits/89337e2553cb74ab31602a4295fa4281cebdb9e0).

File: src/web/tools/InteractionManager.ts (hunk 6 above)
Comment (m-bert) [reply src/web/tools/InteractionManager.ts]: Done in [c392a94](https://github.com/software-mansion/react-native-gesture-handler/pull/2783/commits/c392a94089df0e392f4f910682b4154da4504fbb).

**Pull request merged**